File: 0459-Repeated-Substring-Pattern/0459.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        n = len(s)

        # Try each possible substring length
        for len_sub in range(1, n // 2 + 1):
            # Check if n is divisible by len_sub
            if n % len_sub == 0:
                # Extract the substring and repeat it to match s
                sub = s[:len_sub]
                repeat = sub * (n // len_sub)

                # Check if this repeated string equals s
                if repeat == s:
                    return True
        
        return False
    
    def repeatedSubstringPattern1(self, s: str) -> bool:
        new_s = (s + s)[1:-1]
        return s in new_s

File: 0459-Repeated-Substring-Pattern/0459.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        int n = s.length();

        for (int i = 1; i < (n / 2) + 1; i++) {
            if (n % i == 0) {
                String sub = s.substring(0, i);

                StringBuilder repeat = new StringBuilder();
                int count = n / i;

                for (int j = 0; j < count; j++) {
                    repeat.append(sub);
                }

                if (repeat.toString().equals(s))
                    return true;
            }
        }

        return false;
    }

    public boolean repeatedSubstringPattern1(String s) {
        String doubleS = s + s;
        String newS = doubleS.substring(1, doubleS.length() - 1);

        return newS.contains(s);
    }
}